src/Hosting/Server/ServerList.php

```diff
@@ -105,6 +106,12 @@ public function getLowestNumberedServer()
         $lowestServer = null;
         $lowestId = 999999;
 
+        if (!$this->count()) {
+            throw new RuntimeException(
```

Comment:
Just spitballing... would it be useful to define a custom error class (e.g. `EmptyServerListException`) to make it easier to catch this specific case in code that calls this method?